### PR TITLE
Fix start time of phase "running"

### PR DESCRIPTION
**Pull request by rinckm**
### Description

The start time of phase "running" was set to the start time of the Tekton TaskRun for JFR, which is when the pod has been _created_. But phase "waiting" now covers the time until successful start-up of the containers in the pod, which can be significantly after pod creation, e.g. due to image pull time.

Therefore, the start time of phase "running" is now the start time of the the JRF container.

### Submitter checklist

- [ ] Change has been tested (on a back-end cluster)
- [x] [changelog.yaml] entry with upgrade notes is prepared and appropriate for the audience affected by the change (users or developer, depending on the change)
- [x] Semantic version diffed against [last release][releases] and updated accordingly. In this project the version has to be maintained here:
    - [/charts/steward/Chart.yaml](https://github.com/SAP/stewardci-core/blob/master/charts/steward/Chart.yaml) (`version` and `appVersion`)

In case dependencies have been updated:
- [ ] Links to external changelogs, since the last release of our component, added to the [changelog.yaml] entry (description).
- [ ] Changelogs read thoroughly, potential impact described, upgrade notes prepared (if necessary)
- [ ] Check if dependency updates affect our semantic version increment.

### Reviewer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There is at least 1 approval for the pull request and no outstanding requests for change
- [ ] All voter checks have passed
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] The Pull Request title is understandable and reflects the changes well
- [ ] The Pull Request description is understandable and well documented
- [ ] [changelog.yaml] entry for this Pull Request has been added
    - [ ] Changelog entry contains all required information
    - [ ] 'Upgrade notes' are documented in changelog.yaml (if required)

[changelog.yaml]: https://github.com/SAP/stewardci-core/changelog.yaml
[releases]: https://github.com/SAP/stewardci-core/releases

### CODE REVIEW

File: changelog.yaml [hunk 1]
```diff
@@ -52,6 +52,15 @@
   date: TBD
   changes:
 
+    - type: internal
+      impact: patch
+      title: Fix Start Time
+      description: |-
+        The start time of running phase started at tekton task run start time.
+        With this fix we now take the start time of the step executing the jenkins file runner.
+      pullRequestNumber: 0
```
Comment (rinckm) [inline changelog.yaml:61]:
```suggestion
      pullRequestNumber: 350
```

File: pkg/runctl/run.go [hunk 2]
```diff
@@ -30,7 +34,12 @@ func (r *tektonRun) GetStartTime() *metav1.Time {
 	if condition.IsUnknown() && condition.Reason != tekton.TaskRunReasonRunning.String() {
 		return nil
 	}
-	return r.tektonTaskRun.Status.StartTime
+	for _, step := range r.tektonTaskRun.Status.Steps {
+		if step.ContainerName == jfrStepName && step.Running != nil {
+			return &step.Running.StartedAt
+		}
```
Comment (anbrsap) [inline pkg/runctl/run.go:40]: The container might have terminated already, in which case the start timestamp is in `step.Terminated.StartedAt` (see [here][1] and [here][2]).

[1]: https://github.com/tektoncd/pipeline/blob/e38d112e752ebbe97e4e3eccb1c271bbe5ecbcf7/vendor/k8s.io/api/core/v1/types.go#L2572-L2585
[2]: https://github.com/tektoncd/pipeline/blob/e38d112e752ebbe97e4e3eccb1c271bbe5ecbcf7/vendor/k8s.io/api/core/v1/types.go#L2549

File: pkg/runctl/run_test.go [hunk 3]
```diff
@@ -32,13 +32,13 @@ const (
     status: Unknown
     type: Succeeded
   podName: build-pod-38aa76
-  startTime: "` + time1 + `"
+  startTime: "2019-05-14T08:24:08Z"
```
Comment (anbrsap) [inline pkg/runctl/run_test.go:35]: Put this into a constant, too. Also name the constants according to their meaning.

File: pkg/runctl/controller_test.go [hunk 4]
```diff
@@ -123,6 +124,16 @@ func Test_Controller_Running(t *testing.T) {
 	assert.Equal(t, api.StateRunning, status.State)
 }
 
+func stepsWithRunningContainer(startTime metav1.Time) []tekton.StepState {
```
Comment (anbrsap) [inline pkg/runctl/controller_test.go:127]: There should also be a test with terminated container.

**Review by anbrsap — APPROVED** [review]

**Pull request merged**